src/indexer/merger.rs

```diff
@@ -56,7 +58,37 @@ fn compute_total_num_tokens(readers: &[SegmentReader], field: Field) -> crate::R
             .sum::<u64>())
 }
 
+/// `ReaderWithOrdinal` is used to be able to easier associate
+/// data with a `SegmentReader`. The ordinal is supposed to be
+/// used as an index access.
+///
+/// The ordinal identifies the position within `Merger` readers.
+#[derive(Clone, Copy)]
+pub struct SegmentReaderWithOrdinal<'a> {
+    reader: &'a SegmentReader,
```

Comment:
nitpick: The access is not helping, and once this struct is private, these two could be `pub` member without accessors.

Reply:
yes that's better